pkg/snet/network.go

```diff
@@ -252,6 +281,22 @@ func (n *Network) IsNetworkReady(netType string) bool {
 	return ok
 }
 
+// WaitTillNetworkReady waits till any one of the network is ready.
+func (n *Network) WaitTillNetworkReady(log *logging.Logger) {
+	n.netsMu.Lock()
```

Comment:
I would use a done channel and sync.Once instead of active polling

Reply:
Good idea. didn't think of it.